sonarts-sq-plugin/sonar-typescript-plugin/src/test/java/org/sonar/plugin/typescript/externalreport/TslintRulesTest.java

```diff
@@ -0,0 +1,366 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugin.typescript.externalreport;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TslintRulesTest {
```

Comment:
it's only rules metadata, do we really need such extensive testing? I would like to simplify it

Reply:
Because we have some redundancy in the way we describe rules (README, typescript, json). I found several differences in sonarts repository, like you can see in the PR.
And yesterday, after a rebase on master, a test asked me to add the following two blocks:
```
  {
    "key": "no-inverted-boolean-check",
    "name": "Boolean checks should not be inverted",
    "type": "CODE_SMELL",
    "url": "https://github.com/SonarSource/SonarTS/blob/master/sonarts-core/docs/rules/no-inverted-boolean-check.md"
  },
  {
    "key": "prefer-optional",
    "name": "Optional property declarations should use \u0027?\u0027 syntax",
    "type": "CODE_SMELL",
    "url": "https://github.com/SonarSource/SonarTS/blob/master/sonarts-core/docs/rules/prefer-optional.md"
  },
```
Each time, you create a new rule, you will just have to copy an past a new block in `org/sonar/l10n/typescript/rules/tslint-sonarts/rules.json`